pkg/cluster/cluster.go

```diff
@@ -365,16 +365,15 @@ func (c *Cluster) Update(e *spec.EtcdCluster) {
 }
 
 func (c *Cluster) delete() {
-	if err := c.gc.CollectCluster(c.cluster.Name, garbagecollection.NullUID); err != nil {
-		c.logger.Errorf("cluster delete: fail to clean up resources %v", err)
-	}
-
 	if c.bm != nil {
-		err := c.bm.cleanup()
-		if err != nil {
+		if err := c.bm.cleanup(); err != nil {
 			c.logger.Errorf("cluster deletion: backup manager failed to cleanup: %v", err)
 		}
 	}
+
+	if err := c.gc.CollectCluster(c.cluster.Name, garbagecollection.NullUID); err != nil {
```

Comment:
Probably not move this down here.
We should delete backup RS first before deleting pvc